nav2_costmap_2d/src/clear_costmap_service.cpp

```diff
@@ -43,18 +43,22 @@ ClearCostmapService::ClearCostmapService(
 
   node_->get_parameter("clearable_layers", clearable_layers_);
 
+  // create fully qualified name
+  std::string full_name(std::string(node_->get_namespace()) +
+    std::string(node_->get_name()));
+
   clear_except_service_ = node_->create_service<ClearExceptRegion>(
-    "clear_except_" + costmap_.getName(),
+    full_name + "clear_except_" + costmap_.getName(),
```

Comment:
It thought these services were already launched in their namespace ei `/global_costmap/clear_entirely_global_costmap`, as that's also what's called out in navigate behavior treees
```
navigate_w_replanning_and_recovery.xml
```

the namespace + node name would make it what exactly?

It looks to be like `/global_costmap/gobal_costmapclear_except_global_costmap` (since you're adding `full_name` without another `_`)

Reply:
When I run on my system, latest ros2, master branch, without this, the namespace and name (ex: global_costmap) isn't there, this is just adding those back. I think somewhere along the line the namespace has been lost. I have been seeing errors where the stack hangs saying:
`12: [bt_navigator-10] [INFO] [bt_navigator]: Waiting for "/local_costmap/clear_entirely_local_costmap" service`


Try pulling ros2 and master branch, launch the stack and then do a `ros2 service list`

I see these services:
```
/clear_around_global_costmap
/clear_around_local_costmap
/clear_entirely_global_costmap
/clear_entirely_local_costmap
/clear_except_global_costmap
/clear_except_local_costmap
```
Note they are missing the namespace and node name.

Reply:
That's super strange, they were definitely in the namespace beforehand, that's how things have been working. Maybe something upstream changed :man_shrugging:

Reply:
ill also take a look tomorrow with my cached ros2 workspace to see if it works as expected and then update it to see if then broken

Reply:
@mkhansen-intel I think the source of the problem is that this:

https://github.com/ros-planning/navigation2/blob/6993fcd950a871f15dccacc565e0134fff8859b1/nav2_costmap_2d/src/costmap_2d_ros.cpp#L62

should be this:
```cpp
rclcpp::NodeOptions().arguments({"--ros-args" + std::string("__ns:=") + absolute_namespace + "--"}))
```
That might fix the problem without changes to `ClearCostmapService`.

In addition, maybe not an issue right now, the service names defined here:

https://github.com/ros-planning/navigation2/blob/6993fcd950a871f15dccacc565e0134fff8859b1/nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml#L18

Should be made relative, i.e. remove the leading '/'

Reply:
Also, it seems to me that `rclcpp::NodeOptions()` should add the required flags. Something to discuss upstream.

Reply:
> Should be made relative

Relative to what? Not `nav2_bt_navigator` surely?

Reply:
@orduno - I agree it seems silly we have to add the "--ros-args -r" in every call to NodeOptions()

Those should only be required in launch files and command line 'ros2 run' invocations.

Reply:
I backed this change out and fixed it by adding the "-r" flag, marking resolved